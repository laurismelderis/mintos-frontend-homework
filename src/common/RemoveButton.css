.remove-button {
  width: 1.5em;
  height: 1.5em;
  margin: 0;
  padding: 0;
  top: -6px;
  right: -6px;
  font-size: 12px;
  position: absolute;
  border: 2px solid white;
  border-radius: 1em;
  background: #333333;
  color: white;
  cursor: pointer;
}

.remove-button:hover {
  width: 1.5em;
  height: 1.5em;
  margin: 0;
  padding: 0;
  top: -6px;
  right: -6px;
  font-size: 12px;
  position: absolute;
  border: 2px solid #333333;
  border-radius: 1em;
  background: white;
  color: #333333;
  cursor: pointer;
  font-weight: 600;
}
