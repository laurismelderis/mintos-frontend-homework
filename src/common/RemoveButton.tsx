import React from 'react'
import './RemoveButton.css'

interface RemoveButtonProps {
  onClick: () => void
}

const RemoveButton = ({ onClick }: RemoveButtonProps) => {
  return (
    <button className='remove-button' type='button' onClick={onClick}>
      x
    </button>
  )
}

export default RemoveButton
