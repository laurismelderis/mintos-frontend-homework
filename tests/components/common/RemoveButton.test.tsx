import React from 'react'
import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import RemoveButton from '../../../src/common/RemoveButton'

describe('RemoveButton', () => {
  it('should render', () => {
    render(<RemoveButton />)

    expect(screen.getByTestId('remove-button')).toBeInTheDocument()
  })

  it('should render correctly', () => {
    const tree = renderer.create(<RemoveButton />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
