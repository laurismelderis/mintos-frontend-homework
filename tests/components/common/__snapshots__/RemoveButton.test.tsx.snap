// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoveButton should render correctly 1`] = `
<button
  className="remove-button"
  data-testid="remove-button"
  type="button"
>
  x
</button>
`;
